dace/libraries/standard/reduction_planner.py

```diff
@@ -0,0 +1,209 @@
+# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
+""" Helper function to compute GPU schedule for reduction node "auto" expansion. """
+
+from dace import dtypes
+from dace.data import Array
+from typing import List
+
+
+def get_reduction_schedule(in_array: Array, axes: List[int]):
+    """
+    Computes a GPU reduction schedule depending on the input data's shape and the axes to reduce
+
+    :in_array: DaCe array describing the input data
+    :axes: List of all the axes to reduce
+    """
+
+    # apply vectorization or not
+    use_vectorization = True
+    # apply mini-warps optimitation or not
+    use_mini_warps = True
+
+    # define the warp size
+    warp_size = 32
+    wide_load_size = 16  #bytes
+
+    schedule = {
```

Comment:
You could maybe add a data class to help with readability a bit:
```python
from dace.frontend.python.replacements import Size
@dataclasses.dataclass
class ReductionSchedule:
  grid: List[Size]
  ...
```

Reply:
Done